### PR TITLE
[tests-only] refactor drone definition

### PR DESCRIPTION
**Pull request by wkloucek**
Things changed:
- declare volumes once and use them like "named volumes"
- sanity check: issue warning if pipeline or step name exceeds 50 characters
- sanity check: issue warning if volume is declared in step but not in pipeline
- sanity check: issue warning if depends_on is referencing non existing pipeline
- helper: print count of all used docker images and versions
- add helper for depends_on mangement: "pipelineDependsOn" and "pipelinesDependsOn"

### CODE REVIEW

**Comment by wkloucek** [comment]
Also consider this: https://github.com/owncloud/core/blob/621a09501d9afd81afcc5f62765b9858ed5a7cce/.drone.star#L1144

File: .drone.star [hunk 1]
```diff
@@ -1551,22 +1464,15 @@ def ocisServer(storage, accounts_hash_difficulty = 4):
   return [
     {
       'name': 'ocis-server',
-      'image': 'webhippie/golang:1.14',
+      'image': 'owncloudci/alpine:latest',
```
Comment (phil-davis) [inline .drone.star:1467]: That's "magic". I guess we can run the binary `ocis/bin/ocis` without having to install golang.

File: .drone.star [hunk 2]
```diff
@@ -1724,3 +1622,71 @@ def rebuildBuildArtifactCache(ctx, name, path):
...
+def pipelineSanityChecks(ctx, pipelines):
+  """pipelineSanityChecks helps the CI developers to find errors before running it
+
+  These sanity checks are only executed on when converting starlark to yaml.
+  Error outputs are only visible when the conversion is done with the drone cli.
+
+  Args:
+    ctx: drone passes a context with information which the pipeline can be adapted to
+    pipelines: pipelines to be checked, no
+    rmally you should run this on the return value of main()
```
Comment (phil-davis) [inline .drone.star:1635]:
```suggestion
    pipelines: pipelines to be checked, normally you should run this on the return value of main()
```

**Review by phil-davis — COMMENTED** [review]
Looks good - I only found one strange break of text of the word "normally"!
Was there a reason for that?

File: .drone.star (hunk 2 above)
Comment (phil-davis) [reply .drone.star:1635]: I will push this change and squash...

**Review by phil-davis — APPROVED** [review]

**Pull request merged**